widgets/src/main/java/org/pentaho/gwt/widgets/client/buttons/ImageButton.java

```diff
@@ -236,6 +240,7 @@ public void setEnabled( boolean isEnabled ) {
       this.setSrc( disabledUrl );
     }
     this.updateStyles();
+    this.updateTabIndex();
```

Comment:
In the original, line 181 does `setTabIndex(0);`, which you did not remove.
Also, with this constructor, the image is always enabled by default and as such the proposed change does not seem to change anything.

Reply:
Removed `setTabIndex(0);`.

Reply:
You did not answer as to why you need this change, as, it seems, it changes nothing.

Reply:
This change was done to make disabled buttons inaccessible.

Reply:
![image](https://github.com/pentaho/pentaho-commons-gwt-modules/assets/113503432/5448ea41-9ac0-4d5d-b0a5-44aa1202ae8d)
For the buttons shown above, the second constructor i.e. line 181, is being called. So as to make the disabled buttons (which had tabIndex set as 0) inaccessible, we created a new method called `updateTabIndex()` and called it in line 242.
`updateTabIndex` will take care of setting tabIndex as 0 or -1 based on the boolean property `isEnabled`.

Reply:
I'm sorry. I misread the compact diff, making me think that the new statement `this.updateTabIndex()` was inside the constructor of line 181... (what I wrote should make sense if read in that light).